api/admin/controller/collection_settings.py

```diff
@@ -334,16 +343,24 @@ def process_settings(self, settings, collection):
 
                 if isinstance(external_integration_link, ProblemDetail):
                     return external_integration_link
-            elif key in flask.request.form:
+            elif value is not None:
                 # Only if the key was present in the request should we add it
                 collection_settings[key] = value
+            else:
+                # Keep existing setting value, when present, if a value is not specified.
+                # This can help prevent accidental loss of settings due to some programming errors.
+                if key in collection.integration_configuration.settings_dict:
+                    collection_settings[
+                        key
+                    ] = collection.integration_configuration.settings_dict[key]
```

Comment:
Are we reasonably sure that the admin UI always supplies a value for a blank setting? Want to make sure that we are not introducing a bug here where we will refuse the clear empty fields.

Reply:
Good point! I will verify and remove this case, if it does not behave correctly.

Reply:
Verified.